.github/workflows/test.yml

```diff
@@ -43,7 +43,10 @@ jobs:
       - name: Configure and build
         run: |
           conan profile detect
-          conan build . -s:h compiler.cppstd=23 -s:h openql/*:build_type=${{ matrix.build_type }} -o openql/*:build_tests=True -o openql/*:disable_unitary=True -b missing
+          build_type=${{ matrix.build_type }}
+          lowercase_build_type=$(echo ${build_type} | tr '[:upper:]' '[:lower:]')
```

Comment:
I'm wondering like in the previous PR, if we can set `matrix.build_type` to the lowercase version and ask CMake to build in the lowercase build dir, so that this transformation is not needed.

Reply:
Thanks for the comment Pablo. I valued that option together with changing the profile names to start with uppercase, but I finally decided I like both 1) build directories starting with uppercase, and 2) lowercase for the profile names.

BTW, testing that line took me a while. It seems that `tr` is POSIX standard, but things I tried before, like `${build_type@L}` or `${build_type,,}` were not. 😊